### PR TITLE
added "ops" setting to each rule

### LINKED ISSUE #355: ops Permission Level Not in Autocomplete in 1.4
Don't know if you know this or not, but the ops permission level for command settings added in 1.4 is not in autocomplete.

### PR DESCRIPTION
**Pull request by Ghoulboy78**
Fixes #355

### CODE REVIEW

**Comment by gnembon** [comment]
That's not a correct place to put it, add it here:
https://github.com/gnembon/fabric-carpet/blob/3f593943c9d92a2b5ef1bdf9d7b4f87bebd377fe/src/main/java/carpet/settings/ParsedRule.java#L50

**Comment by Ghoulboy78** [comment]
wdym, put it there?

**Comment by gnembon** [comment]
well, this code applies custom operations to rules that are commands and are registered as a string. You can put options to 'true', 'false', 'ops' there.

**Comment by Ghoulboy78** [comment]
how though?

**Comment by Ghoulboy78** [comment]
don't you mean here?
https://github.com/gnembon/fabric-carpet/blob/3f593943c9d92a2b5ef1bdf9d7b4f87bebd377fe/src/main/java/carpet/settings/ParsedRule.java#L64-L67

**Comment by gnembon** [comment]
yes - there exactly

**Comment by Ghoulboy78** [comment]
ok done, just wanted to confirm before committing, cos now I gotta undo all the stuff I did initially

File: src/main/java/carpet/settings/ParsedRule.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
         }
         else if (this.type == boolean.class || (this.type == String.class && categories.contains(RuleCategory.COMMAND)))
```
Comment (gnembon) [inline src/main/java/carpet/settings/ParsedRule.java:64]: if its a boolean type, then do not add 'ops'. Add ops only if its string class and command

File: src/main/java/carpet/CarpetSettings.java [hunk 2]
```diff
@@ -652,4 +650,4 @@ public static void changeSpawnSize(int size)
             validate = Validator.PROBABILITY.class
     )
     public static double creativeFlyDrag = 0.09;
-}
+}
```
Comment (gnembon) [inline src/main/java/carpet/CarpetSettings.java:653]: while at it, you might revert all the changes to CarpetSettings - not to introduce unnecessary diffs.

Comment (Ghoulboy78) [reply src/main/java/carpet/CarpetSettings.java]: done

File: src/main/java/carpet/settings/ParsedRule.java (hunk 1 above)
Comment (Ghoulboy78) [reply src/main/java/carpet/settings/ParsedRule.java]: done

**Pull request merged**